core/src/main/scala/za/co/absa/spline/harvester/builder/plan/MergeIntoNodeBuilder.scala

```diff
@@ -29,7 +29,14 @@ class MergeIntoNodeBuilder
   (idGenerators: IdGeneratorsBundle, dataTypeConverter: DataTypeConverter, dataConverter: DataConverter, postProcessor: PostProcessor)
   extends GenericPlanNodeBuilder(logicalPlan)(idGenerators, dataTypeConverter, dataConverter, postProcessor) {
 
-  private lazy val mergeInputs: Seq[Seq[Attribute]] = inputAttributes.transpose
+  private lazy val mergeInputs: Seq[Seq[Attribute]] = {
+    val Seq(srcAttrs, trgAttrs) = inputAttributes
+    val srcAttrsByName = srcAttrs.map(a => a.name -> a).toMap
+    trgAttrs.map(trg => {
+      val src = srcAttrsByName.get(trg.name)
+      Seq(trg) ++ src
+    })
+  }
```

Comment:
What happens when the names don't match:
```
  | WHEN MATCHED THEN
  |   UPDATE SET
  |     name = src.full_name
```

I would rename `src` -> `maybeSrc`

Reply:
> I would also rename `src` -> `maybeSrc`

renamed

> What happens when the names don't match

Well, I guess nothing. Since there is no reference from a target attribute to the expression in Spark exec plan (or at least I can't see it) we cannot do much about it. For example from the below screenshot how would I infer dependencies between `full_name -> name` or `code -> name`? The agent could guess based on e.g. Levenshtein distance, but that would be another feature.

![image](https://user-images.githubusercontent.com/795479/219596063-11b17130-b76d-4f54-bf29-1bd4de06e613.png)

Actually, now thinking about it more I realized that even if names match it doesn't mean there is a dependency as there could be mutual renaming in the query (`UPDATE SET dst.a = src.b, dst.b = src.a`).

Do you have any better suggestion how to do it properly?

Reply:
> Actually, now thinking about it more I realized that even if names match it doesn't mean there is a dependency as there could be mutual renaming in the query (UPDATE SET dst.a = src.b, dst.b = src.a).

Exactly. I think the only way to do this properly is to get the attribute pairing from UPDATE SET condition.